src/main/java/it/gov/pagopa/pu/debtpositions/service/massive/action/InsertionActionMassiveDebtPositionServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package it.gov.pagopa.pu.debtpositions.service.massive.action;
+
+import it.gov.pagopa.pu.debtpositions.dto.generated.*;
+import it.gov.pagopa.pu.debtpositions.exception.custom.ConflictErrorException;
+import it.gov.pagopa.pu.debtpositions.mapper.DebtPositionMapper;
+import it.gov.pagopa.pu.debtpositions.model.DebtPosition;
+import it.gov.pagopa.pu.debtpositions.model.InstallmentNoPII;
+import it.gov.pagopa.pu.debtpositions.repository.InstallmentNoPIIRepository;
+import it.gov.pagopa.pu.debtpositions.service.create.debtposition.CreateDebtPositionService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@Slf4j
+public class InsertionActionMassiveDebtPositionServiceImpl implements InsertionActionMassiveDebtPositionService {
+
+    private final InstallmentNoPIIRepository installmentNoPIIRepository;
+    private final CreateDebtPositionService createDebtPositionService;
+    private final DebtPositionMapper debtPositionMapper;
+
+    private static final Set<InstallmentStatus> installmentStatusesValidForInsertion = Set.of(InstallmentStatus.CANCELLED, InstallmentStatus.INVALID);
+    private static final Set<PaymentOptionStatus> paymentOptionStatusesValidForInsertion = Set.of(PaymentOptionStatus.UNPAID, PaymentOptionStatus.EXPIRED, PaymentOptionStatus.PARTIALLY_PAID);
+    private static final Set<DebtPositionStatus> debtPositionStatusesValidForInsertion = Set.of(DebtPositionStatus.UNPAID, DebtPositionStatus.EXPIRED, DebtPositionStatus.PARTIALLY_PAID);
+
+    public InsertionActionMassiveDebtPositionServiceImpl(InstallmentNoPIIRepository installmentNoPIIRepository, CreateDebtPositionService createDebtPositionService, DebtPositionMapper debtPositionMapper) {
+        this.installmentNoPIIRepository = installmentNoPIIRepository;
+        this.createDebtPositionService = createDebtPositionService;
+        this.debtPositionMapper = debtPositionMapper;
+    }
+
+    @Override
+    public String handleInsertion(DebtPositionDTO debtPositionSynchronizeDTO, DebtPosition debtPosition, Boolean massive, String accessToken, String operatorExternalUserId) {
+        Optional<InstallmentNoPII> installment = installmentNoPIIRepository.getByOrganizationIdAndIudAndPaymentOptionIndexAndIuv(
```

Comment:
As first things, if the debtPosition doesnt' exists, we could invoke and return the createInstallment method (it will perform also unique id check)